### ISSUE TITLE
Using fluent-plugin-elasticsearch `v.4.1.2` and above does not work with Elasticsearch `v.7.7.1`

### ISSUE THREAD

**Issue by adnanrahic** [uken/fluent-plugin-elasticsearch#854]
(check apply)
- [x] read [the contribution guideline](https://github.com/uken/fluent-plugin-elasticsearch/blob/master/CONTRIBUTING.md)
- [x] (optional) already reported 3rd party upstream repository or mailing list if you use k8s addon or helm charts.

#### Problem
While following [this guide](https://docs.fluentd.org/how-to-guides/logs-to-sematext), log shipping does not work and ends up with this error:

```
[error]: #0 unexpected error error_class=NoMethodError error="undefined method `dig' for #<String:0x0000557b648ef9d8>"
```

#### Steps to replicate

I used this config:

```
# Switch to debug if you need to debug
<system>
  log_level debug
</system>

# get logs from syslog
<source>
  @type syslog
  port 42185
  tag syslog
</source>

# get logs from fluent-logger, fluent-cat or other Fluentd instances
<source>
  @type forward
</source>

<filter **>
  @type stdout
</filter>

<match syslog.**>
  @type elasticsearch
  host logsene-receiver.sematext.com
  # for EU
  # host logsene-receiver.eu.sematext.com
  port 443
  scheme https
  index_name <LOGS_TOKEN>
  <buffer>
    @type file
    path /tmp/fluent/es-buffer/es.all.*.buffer
    chunk_limit_size 250k
    flush_interval 50s
    flush_thread_count 1
    retry_limit 5
    retry_wait 60
  </buffer>
</match>
```

The systemd service starts, but log shipping does not work, and shows this error in `/var/log/td-agent/td-agent.log`:

```
[error]: #0 unexpected error error_class=NoMethodError error="undefined method `dig' for #<String:0x0000557b648ef9d8>"
```

#### Expected Behavior or What you need to ask
Log shipping should work.

After testing with other versions of both Elasticsearch and fluent-plugin-elasticsearch I noticed this issue it NOT present with v.4.1.1 of fluent-plugin-elasticsearch and Elasticsearch v.7.7.1.

The issue is NOT present with Elasticsearch v.7.4.0 and fluent-plugin-elasticsearch v.4.1.2 and above.

Do you know why is this happening? :thinking: 

#### Using Fluentd and ES plugin versions

Not working:

* OS: Ubuntu 18.4
* VM
* Agent: td-agent 1.11.2
* ES plugin >4.1.2
* ES version 7.7.1

Working:

* OS: Ubuntu 18.4
* VM
* Agent: td-agent 1.11.2
* ES plugin 4.1.1
* ES version 7.7.1

OR

* OS: Ubuntu 18.4
* VM
* Agent: td-agent 1.11.2
* ES plugin >4.1.2
* ES version 7.4.0

**Comment by cosmo0920** [comment]
What is the result to execute the following commands?

```ruby
require 'elasticsearch'

client = Elasticsearch::Client.new
puts client.info
puts client.info.class
```

**Comment by cosmo0920** [comment]
Patch is here: https://github.com/uken/fluent-plugin-elasticsearch/pull/855

**Comment by adnanrahic** [comment]
Hi @cosmo0920! Thanks for looking into this. Is there anything else you'd need from me?
Or, will patch #855 be merged?

**Comment by cosmo0920** [comment]
#833 is merged.  It will be released soon.

**Comment by cosmo0920** [comment]
#855 is released as v4.3.3.
